source/extensions/filters/http/rate_limit_quota/quota_bucket_cache.h

```diff
@@ -46,6 +46,8 @@ struct Bucket {
   QuotaUsage quota_usage;
   // Rate limiter based on token bucket algorithm.
   TokenBucketPtr token_bucket_limiter;
+  // First assignment time.
+  std::chrono::nanoseconds first_assignment_time = {};
```

Comment:
For clarity it should be Envoy::MonotonicTime since it specifies a point in time.